data-prepper-plugins/kinesis-source/src/test/java/org/opensearch/dataprepper/plugins/kinesis/source/processor/KinesisRecordProcessorTest.java

```diff
@@ -0,0 +1,371 @@
+    private PluginFactory pluginFactory;
+
+    @Mock
+    private KinesisSourceConfig kinesisSourceConfig;
+
+    @Mock
+    private KinesisStreamConfig kinesisStreamConfig;
+
+    @Mock
+    private InitializationInput initializationInput;
+
+    @Mock
+    private ProcessRecordsInput processRecordsInput;
+
+    @Mock
+    private RecordProcessorCheckpointer checkpointer;
+
+    @Mock
+    private Buffer<Record<Event>> buffer;
+
+    @Mock
+    StreamIdentifier streamIdentifier;
+
+    @Mock
+    private AcknowledgementSetManager acknowledgementSetManager;
+
+    @Mock
+    private AcknowledgementSet acknowledgementSet;
+
+    @Mock
+    private Counter recordProcessingErrors;
+
+    @Mock
+    private Counter checkpointFailures;
+
+    @Mock
+    private Counter acknowledgementSetCallbackCounter;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        pluginMetrics = mock(PluginMetrics.class);
+        pluginFactory = mock(PluginFactory.class);
+        acknowledgementSet = mock(AcknowledgementSet.class);
+
+        when(initializationInput.shardId()).thenReturn(shardId);
+        when(streamIdentifier.streamName()).thenReturn(streamId);
+        when(kinesisStreamConfig.getName()).thenReturn(streamId);
+        PluginModel pluginModel = mock(PluginModel.class);
+        when(pluginModel.getPluginName()).thenReturn(codec_plugin_name);
+        when(pluginModel.getPluginSettings()).thenReturn(Collections.emptyMap());
+        when(kinesisSourceConfig.getCodec()).thenReturn(pluginModel);
+
+        InputCodec codec = mock(InputCodec.class);
+        when(pluginFactory.loadPlugin(eq(InputCodec.class), any())).thenReturn(codec);
+        when(kinesisStreamConfig.getCheckPointInterval()).thenReturn(CHECKPOINT_INTERVAL);
+        when(kinesisSourceConfig.getNumberOfRecordsToAccumulate()).thenReturn(NUMBER_OF_RECORDS_TO_ACCUMULATE);
+        when(kinesisSourceConfig.getStreams()).thenReturn(List.of(kinesisStreamConfig));
+        when(processRecordsInput.checkpointer()).thenReturn(checkpointer);
+        when(pluginMetrics.counterWithTags(ACKNOWLEDGEMENT_SET_CALLBACK_METRIC_NAME, KINESIS_STREAM_TAG_KEY, streamIdentifier.streamName())).thenReturn(acknowledgementSetCallbackCounter);
+    }
+
+    @Test
+    void testProcessRecordsWithoutAcknowledgementsCheckpointsEnabled()
+            throws Exception {
+        when(processRecordsInput.records()).thenReturn(createInputKinesisClientRecords());
+        when(kinesisSourceConfig.isAcknowledgments()).thenReturn(false);
+        when(kinesisStreamConfig.isEnableCheckPoint()).thenReturn(true);
+        when(acknowledgementSetManager.create(any(), any(Duration.class))).thenReturn(acknowledgementSet);
+
+        kinesisRecordProcessor = new KinesisRecordProcessor(buffer, kinesisSourceConfig, acknowledgementSetManager, pluginMetrics, pluginFactory, streamIdentifier);
+        kinesisRecordProcessor.initialize(initializationInput);
+
+        Thread.sleep(2000);
```

Comment:
Use Awaitility instead of a sleep.

Reply:
I have modified the test to remove this.